test/fail_compilation/retscope6.d

```diff
@@ -21,3 +21,55 @@ int* test() @safe
     arr[0] ~= &i;
     return arr[0][0];
 }
+
+/* TEST_OUTPUT:
+---
+fail_compilation/retscope6.d(7034): Error: reference to local variable `i` assigned to non-scope parameter `_param_1` calling retscope6.S.emplace!(int*).emplace
+fail_compilation/retscope6.d(7035): Error: reference to local variable `i` assigned to non-scope parameter `_param_0` calling retscope6.S.emplace2!(int*).emplace2
+fail_compilation/retscope6.d(7024): Error: scope variable `_param_2` assigned to `s` with longer lifetime
+fail_compilation/retscope6.d(7025): Error: scope variable `_param_2` assigned to `t` with longer lifetime
+fail_compilation/retscope6.d(7037): Error: template instance `retscope6.S.emplace4!(int*)` error instantiating
```

Comment:
I recommend a single section of test output, it will be much easier if the error messages need to be updated. It caused me quite some pain in another PR.

Reply:
I do multiple blocks with a `#line` precisely because they are easier to update. Adding a line doesn't require updating every message line.

Reply:
`./run.d fail_compilation AUTO_UPDATE=1` for the win...

Reply:
Yeah, this only works with a single output section.